### PR TITLE
Clean up inflation swaps convention

### PR DESCRIPTION
**Pull request by brianweller89**
- USCPI now interpolated
- JPY now interpolated (using 10th of the month inflation base)
- All GBP indices now use 2 month inflation lag.

Primary Source: [Barclays Inflation Derivatives Users Guide](http://www.trade2win.com/boards/attachments/next-steps/101520d1296424640-inflation-trading-barclays-inflation-derivatives-users-guide.pdf)

_“The trading month for UK RPI breakeven swaps has a lag of two months“_

_“US zero coupon CPI swaps have adopted an interpolated base index format“_

_“…with the most standard format for [JPY] zero coupon breakevens using the same interpolated inflation basis as JGBi bonds, ie, a three-month lagged inflation reference from the tenth of the month. “_

### CODE REVIEW

File: modules/product/src/main/java/com/opengamma/strata/product/swap/type/InflationRateSwapLegConvention.java [hunk 1]
```diff
@@ -121,15 +121,17 @@
...
    * @param index  the index, the market convention values are extracted from the index
    * @param lag  the lag between the price index and the accrual date, typically a number of months
+   * @param priceIndexCalculationMethod the price index calculation method, typically interpolated or monthly
    * @param businessDayAdjustment the business day 
    * @return the convention
    */
   public static InflationRateSwapLegConvention of(
       PriceIndex index,
       Period lag,
+      PriceIndexCalculationMethod priceIndexCalculationMethod,
```
Comment (jodastephen) [inline modules/product/src/main/java/com/opengamma/strata/product/swap/type/InflationRateSwapLegConvention.java]: This is an incompatible change. Need to add an overload.

File: modules/product/src/main/java/com/opengamma/strata/product/swap/type/InflationRateSwapLegConvention.java [hunk 2]
```diff
@@ -121,15 +121,17 @@
...
       BusinessDayAdjustment businessDayAdjustment) {
 
-    return new InflationRateSwapLegConvention(index, lag, PriceIndexCalculationMethod.MONTHLY, false,
+    return new InflationRateSwapLegConvention(index, lag, priceIndexCalculationMethod, false,
```
Comment (jodastephen) [inline modules/product/src/main/java/com/opengamma/strata/product/swap/type/InflationRateSwapLegConvention.java]: Each argument should be on a new line

File: modules/product/src/main/java/com/opengamma/strata/product/swap/type/StandardFixedInflationSwapConventions.java [hunk 3]
```diff
@@ -157,6 +194,7 @@ private static FixedRateSwapLegConvention fixedLegZcConvention(Currency ccy, Hol
   }
 
   //-------------------------------------------------------------------------
+
```
Comment (jodastephen) [inline modules/product/src/main/java/com/opengamma/strata/product/swap/type/StandardFixedInflationSwapConventions.java]: No blank line here

File: modules/product/src/test/java/com/opengamma/strata/product/swap/type/FixedInflationSwapConventionTest.java [hunk 4]
```diff
@@ -56,9 +57,9 @@
   private static final FixedRateSwapLegConvention FIXED = fixedLegZcConvention(GBP, GBLO);
   private static final FixedRateSwapLegConvention FIXED2 =
       FixedRateSwapLegConvention.of(GBP, ACT_365F, P3M, BDA_MOD_FOLLOW);
-  private static final InflationRateSwapLegConvention INFL = InflationRateSwapLegConvention.of(GB_HICP, LAG_3M, BDA_MOD_FOLLOW);
-  private static final InflationRateSwapLegConvention INFL2 = InflationRateSwapLegConvention.of(GB_RPI, LAG_3M, BDA_MOD_FOLLOW);
-  private static final InflationRateSwapLegConvention INFL3 = InflationRateSwapLegConvention.of(GB_RPIX, LAG_3M, BDA_MOD_FOLLOW);
+  private static final InflationRateSwapLegConvention INFL = InflationRateSwapLegConvention.of(GB_HICP, LAG_3M, MONTHLY, BDA_MOD_FOLLOW);
+  private static final InflationRateSwapLegConvention INFL2 = InflationRateSwapLegConvention.of(GB_RPI, LAG_3M, MONTHLY, BDA_MOD_FOLLOW);
+  private static final InflationRateSwapLegConvention INFL3 = InflationRateSwapLegConvention.of(GB_RPIX, LAG_3M, MONTHLY, BDA_MOD_FOLLOW);
```
Comment (jodastephen) [inline modules/product/src/test/java/com/opengamma/strata/product/swap/type/FixedInflationSwapConventionTest.java]: Could back this out when compatibility overload added.

File: modules/product/src/main/java/com/opengamma/strata/product/swap/type/InflationRateSwapLegConvention.java (hunk 1 above)
Comment (brianweller89) [inline modules/product/src/main/java/com/opengamma/strata/product/swap/type/InflationRateSwapLegConvention.java]: Ok. Have deprecated the old method as the client should have to provide interpolation style when defining conventions. Defaulting to monthly seems slightly arbitrary.

File: modules/product/src/test/java/com/opengamma/strata/product/swap/type/FixedInflationSwapConventionTest.java (hunk 4 above)
Comment (brianweller89) [inline modules/product/src/test/java/com/opengamma/strata/product/swap/type/FixedInflationSwapConventionTest.java]: Would rather keep the test changes in order to discourage the use of the now deprecated method.

**Pull request merged**